### PR TITLE
Warn teachers on Spritelab image import

### PR DESCRIPTION
**Pull request by bencodeorg**
Add a warning that shows up the first time a teacher tries to upload an image in a Sprite Lab project. This warning is similar to that seen by students (implemented [here](https://github.com/code-dot-org/code-dot-org/pull/50269)), but does not restrict sharing/remixing as we do for students.

image

For now, this warning only shows up if you have the BACKGROUNDS_AND_UPLOAD [experiment flag](https://github.com/code-dot-org/code-dot-org/pull/50310/files#diff-1b812462009f740ac2d7308869034061e47ea3a8bc66c42e815d46801aea4b66R40), `backgroundsTab`, enabled and are in a Blockly lab (ie, Spritelab).

## Links

- jira ticket: [SL-617](https://codedotorg.atlassian.net/browse/SL-617)

## Testing story

Tested manually for both student and teacher scenarios that, with the experiment enabled, teachers see this warning once when they upload an image for the first time (via the "costumes" or "animations" tabs), and do not see it on subsequent uploads. Also confirmed warning was not shown on page refresh (ie, reloading project from JSON). Added unit tests to cover new UI scenarios.

### CODE REVIEW

File: apps/test/unit/p5lab/AnimationPicker/AnimationUploadButtonTest.js [hunk 1]
```diff
@@ -7,84 +7,121 @@ import AnimationPickerListItem from '../../../../src/p5lab/AnimationPicker/Anima
...
-        inRestrictedShareMode={false}
-        showingUploadWarning={emptyFunction}
-        exitedUploadWarning={emptyFunction}
-      />
-    );
-    const uploadButton = body.find(AnimationPickerListItem).at(0);
-    uploadButton.simulate('click');
-    const warningModal = body.find(BaseDialog).at(0);
-    const buttons = warningModal.find('button');
-    console.log(`found ${buttons.length} buttons`);
```
Comment (bencodeorg) [inline apps/test/unit/p5lab/AnimationPicker/AnimationUploadButtonTest.js:81]: @molly-moen removed these console logs as I think they're just leftover from your initial testing, LMK if not

File: apps/src/p5lab/P5LabView.jsx [hunk 2]
```diff
@@ -104,14 +104,24 @@ class P5LabView extends React.Component {
...
+  // Users of non-blockly labs should always be allowed to upload animations
+  // with no restrictions. Teachers in blockly labs (ie, sprite lab) can upload with a warning.
+  // If students upload animations we will disable publish and remix.
+  shouldWarnOnAnimationUpload() {
+    if (!this.props.isBlockly) {
       return false;
     }
 
-    return this.props.isBlockly;
+    // hide warning modal for teachers until we've enabled this experiment for everyone
```
Comment (molly-moen) [inline apps/src/p5lab/P5LabView.jsx]: nit: should this comment be "hide warning modal for teachers if they don't have the experiment enabled"?

File: apps/test/unit/p5lab/AnimationPicker/AnimationUploadButtonTest.js (hunk 1 above)
Comment (molly-moen) [reply apps/test/unit/p5lab/AnimationPicker/AnimationUploadButtonTest.js:81]: yep, good catch!

**Review by molly-moen — APPROVED** [review]
LGTM!

File: apps/src/p5lab/P5LabView.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/p5lab/P5LabView.jsx]: Yes sorry this isn't super clear, will update.

Comment (bencodeorg) [reply apps/src/p5lab/P5LabView.jsx]: Decided code is self-documenting, so removed this confusing comment :)

**Pull request merged**